Source/player.cpp

```diff
@@ -2431,52 +2443,56 @@ void NextPlrLevel(Player &player)
 	CalcPlrInv(player, true);
 }
 
-void AddPlrExperience(Player &player, int lvl, int exp)
+void Player::_addExperience(uint32_t experience, int levelDelta)
 {
-	if (&player != MyPlayer || player._pHitPoints <= 0)
+	if (this != MyPlayer || _pHitPoints <= 0)
+		return;
+
+	const uint8_t MaxCharacterLevel = getMaxCharacterLevel();

```

Comment:
```suggestion
	const uint8_t maxCharacterLevel = getMaxCharacterLevel();
```
local variable and to prevent confusion with the `MaxCharacterLevel` constexpr in playerdat.cpp